src/modules/translators/implementation.nix

```diff
@@ -1,25 +1,33 @@
-{
-  config,
-  callPackageDream,
-  ...
-}: let
-  lib = config.lib;
-  t = lib.types;
-  # TODO: the makeTranslator logic should be moved somewhere to /src/modules
-  loader = (callPackageDream ../../subsystems/translators.nix {}).makeTranslator;
-  funcs = config.functions.subsystem-loading;
+{config, ...}: let
+  funcs = import ../subsystem-loading.nix config;
+
   collectedModules = funcs.collect "translators";
+  importedModules = funcs.import_ collectedModules;
+
+  wrapTranslator = t:
+    t
+    // config.lib.optionalAttrs (t.translate or null != null) {
+      translate = args:
+        t.translate
+        (
+          (config.functions.translators.makeTranslatorDefaultArgs
+            (t.extraArgs or {}))
+          // args
+          // (args.project.subsystemInfo or {})
+          // {
+            tree =
+              args.tree or (config.dlib.prepareSourceTree {inherit (args) source;});
+          }
+        );
+      translateBin =
+        config.functions.translators.wrapPureTranslator
+        {inherit (t) subsystem name;};
+    };
 in {
   config = {
     # The user can add more translators by extending this attribute
-    translators = funcs.import_ collectedModules;
-
-    /*
-    translators wrapped with extra logic to add extra attributes,
-    like .translateBin for pure translators
-    */
-    translatorInstances = funcs.instantiate config.translators loader;
+    translators = config.lib.mapAttrs (_: wrapTranslator) importedModules;
```

Comment:
The problem I see with this, is that the user cannot add translators easily anymore via the module system by adding an attribute `translators.new-translator = ...`. Now the user manually has to take care to call wrapTranslator.

If we have it separated like before, the user could just add to `translators` and `translatorInstances` would always contain the wrapped version of the translator.

Reply:
I forgot to take that into account before I changed it to this so I will just change it back to how it was before. It used an `implementation.nix` in `interfaces.translator` to load these stuff so it would work just with `translators` but that caused infinite recursion errors and I wasn't able to fix it.

Reply:
I was able to find a different solution so I used that.